test/run-test.py

```diff
@@ -0,0 +1,66 @@
+from lakefs_client.model.repository_creation import RepositoryCreation
+import pyspark
+from pyspark.sql import SparkSession
+from pyspark.conf import SparkConf
+
+
+MOCK_EMAIL = "test@acme.co"
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--storage_namespace", default="local://", required=True)
+    parser.add_argument("--repository", default="example")
+    lakefs_access_key = 'AKIAIOSFODNN7EXAMPLE'
+    lakefs_secret_key = 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
+
+    args = parser.parse_args()
+
+    lfs_client = LakeFSClient(
+        lakefs_client.Configuration(username=lakefs_access_key,
+                                    password=lakefs_secret_key,
+                                    host='http://localhost:8000'))
+
+    # Setup lakeFS
+    repo_name = args.repository
+    lfs_client.config.setup_comm_prefs(CommPrefsInput(feature_updates=False, security_updates=False, email=MOCK_EMAIL))
+    lfs_client.config.setup(Setup(username="lynn",
+                                key=AccessKeyCredentials(access_key_id= lakefs_access_key, secret_access_key= lakefs_secret_key)))
+
+    lfs_client.repositories.create_repository(
+        RepositoryCreation(name=repo_name, storage_namespace=args.storage_namespace))
+
+    spark_config = SparkConf()
+    spark_config.set("spark.sql.catalog.lakefs", "org.apache.iceberg.spark.SparkCatalog")
+    spark_config.set("spark.sql.catalog.lakefs.catalog-impl", "io.lakefs.iceberg.LakeFSCatalog")
+    spark_config.set("spark.sql.catalog.lakefs.warehouse", f"lakefs://{repo_name}")
+    spark_config.set("spark.sql.extensions", "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions")
+    spark_config.set("spark.hadoop.fs.s3a.endpoint", "http://localhost:8000")
+    spark_config.set("spark.hadoop.fs.s3a.access.key", lakefs_access_key)
+    spark_config.set("spark.hadoop.fs.s3a.secret.key", lakefs_secret_key)
+    spark_config.set("spark.hadoop.fs.s3a.path.style.access", "true")
+    spark_config.set("spark.jars.packages", "io.lakefs:lakefs-iceberg:1.0-SNAPSHOT,org.apache.iceberg:iceberg-spark-runtime-3.3_2.12:1.3.0,org.apache.hadoop:hadoop-aws:3.3.4,org.apache.hadoop:hadoop-client-api:3.3.4")
+
+    spark = SparkSession.builder.config(conf=spark_config).getOrCreate()
+
+    df = spark.read.option("inferSchema","true").option("multiline","true").csv("./test/data-sets/film_permits.csv")
+    df.write.saveAsTable("lakefs.main.nyc.permits")
+
+    #Commit, create a new branch, check that the tables are the same
+    lfs_client.commits.commit(repo_name, "main", CommitCreation(message="Initial data load"))
+    lfs_client.branches.create_branch(repo_name, BranchCreation(name="dev", source="main"))
+    df_main = spark.read.table("lakefs.main.nyc.permits")
+    df_dev = spark.read.table("lakefs.dev.nyc.permits")
+    assert (df_main.schema == df_dev.schema) and (df_main.collect() == df_dev.collect()), "main and dev tables should be equal"
```

Comment:
In a future PR, can we use some Python testing framework here? To manage the test cases and report the failures.

Reply:
Of course, this is how it will be.